src/components/layerpublisher/LayerPublisher.js

```diff
@@ -43,6 +45,36 @@ function info(fileName) {
   PubSub.publish(STATUS_INFO, `Loaded ${fileName}.`);
 }
 
+async function initRasterLayer(data) {
+  const channelNames = Object.keys(data);
+  const channelUrls = Object.values(data).map(d => d.tileSource);
+  const raster = {
+    channelNames,
+    domains: Object.values(data).map(d => d.range),
+    id: String(Date.now()),
+  };
+  if (channelUrls[0].includes('tif')) {
```

Comment:
This worries me... I think we had said there were no semantics in the URLs? but even if we do want to change our mind on that, shouldn't this be a lot tighter, and only consider the end of the path?

Reply:
This URL logic was from before (#487) and I agree needs to be removed. My goal in this PR was partially to expose and consolidate all the weird, hardcoded logic we have for `raster` sources. This is useful in solving #486. I think this is beyond the scope of this PR, since this was primarily a refactor and does not require a new `vitessce-data` release.

Reply:
This was exactly why I did this in #487.  Under the time-crunch, and also without time to discuss, this felt like the best option.

Reply:
Cool: Didn't know / forgot the history. This is fine. Thanks!